src/model/styles.rs

```diff
@@ -280,9 +288,147 @@ impl Selector {
     pub fn can<T: ?Sized + Any>() -> Self {
         Self::Can(TypeId::of::<T>())
     }
+    pub fn match_iter<'a>(
+        &'a self,
+        introspector: &'a Introspector,
+    ) -> Box<dyn Iterator<Item = (usize, &'a Content)> + 'a> {
+        self.match_iter_inner(introspector, introspector.all().enumerate())
+    }
+
+    /// Calls the methods on this selector.
+    pub fn call_method(
+        &self,
+        vm: &mut Vm,
+        method: &str,
+        selector: Selector,
+        mut args: Args,
+        span: Span,
+        missing: impl FnOnce() -> SourceResult<Value>,
+    ) -> SourceResult<Value> {
+        match method {
+            "before" => {
```

Comment:
I would prefer to have these as actual methods on the `Selector` (which would also be useful in rust code) and then these methods are dispatched in `methods.rs` as usual. (The counter thing is an exception because its in the library.)

Reply:
Sure, I was planning on doing that as I cleanup round anyway, consider it done ;)